value-processor/src/org/immutables/value/processor/meta/NestedBuilderReflection.java

```diff
@@ -0,0 +1,222 @@
+      return true;
+    }
+
+    if (valueAttribute().typeKind().isList()) {
+      System.out.println(valueAttribute().firstTypeParameter());
+      TypeElement unboxedReturnType = valueAttribute()
+          .environment()
+          .processing()
+          .getElementUtils()
+          .getTypeElement(valueAttribute().firstTypeParameter());
+      return unboxedReturnType.getKind() == ElementKind.CLASS
+          || unboxedReturnType.getKind() == ElementKind.INTERFACE;
+    }
+
+    return false;
+  }
+
+  @Lazy
+  public boolean isImmutable() {
+    return unboxedReturnType().asElement().getAnnotation(Immutable.class) != null;
+  }
+
+  /**
+   * @return type mirror for the return type, or the template argument of a single collection
+   * element.
+   */
+  @Lazy
+  protected DeclaredType unboxedReturnType() {
+    if (valueAttribute().typeKind().isRegular()
+        && valueAttribute().returnType.getKind() == TypeKind.DECLARED) {
+      return (DeclaredType) valueAttribute().returnType;
+    } else if (valueAttribute().typeKind().isList()) {
+      TypeElement unboxedReturnType = valueAttribute()
+          .environment()
+          .processing()
+          .getElementUtils()
+          .getTypeElement(valueAttribute().firstTypeParameter());
+
+      if (unboxedReturnType.getKind() == ElementKind.CLASS
+          || unboxedReturnType.getKind() == ElementKind.INTERFACE) {
+        return (DeclaredType) unboxedReturnType.asType();
+      }
+    }
+
+    throw new RuntimeException(
+        String.format("Could not handle valueAttribuet %s", valueAttribute()));
+  }
+
+  @Lazy
+  protected Style.NestedBuilder immutablesNestedBuilderStyle() {
+    return new Style.NestedBuilder() {
+      @Override
+      public Class<? extends Annotation> annotationType() {
+        return Style.NestedBuilder.class;
+      }
+
+      @Override
+      public String getBuilderMethod() {
+        return getNestedImmutableValueType().constitution.style().builder();
+      }
+
+      @Override
+      public String buildMethod() {
+        return getNestedImmutableValueType().constitution.style().build();
+      }
+
+      @Override
+      public FromMethod fromMethod() {
+        return new FromMethod() {
+
+          @Override
+          public Class<? extends Annotation> annotationType() {
+            return FromMethod.class;
+          }
+
+          @Override
+          public Target target() {
+            return Target.BUILDER_INSTANCE;
+          }
+
+          @Override
+          public String methodName() {
+            return getNestedImmutableValueType().constitution.style().from();
+          }
+        };
+      }
+    };
+  }
+
+  @Lazy
+  public ValueType getNestedImmutableValueType() {
+    if (!isImmutable()) {
+      throw new RuntimeException(
+          "You should not be calling getNestedImmutableValueType for non immutable.");
+    }
+
+    ValueType nestedImmutable = new ValueType();
```

Comment:
@elucash this method needs to return the `ValueType` of the nested immutable.

The reason I need the ValueType is to look up the style information.  However, the style information is always the parent style, not the child style.  Is there something different to do here?  Is there a map somewhere with all the different ValueTypes based on a DeclaredType?

I want to make sure the Style follows all the same rules, like if you `@MetaInf` or associate a package with a style, etc.